packages/sourcing/src/LocalFileSystem.ts

```diff
@@ -23,6 +23,27 @@ export default class LocalFileSystem implements FileSystem
         return fs.exists(location);
     }
 
+    // This method is synchronous because it's used in the
+    // LocationRewriter. This class uses a replacer function
```

Comment:
The `LocationRewriter` comes from a different package and has nothing to do with this class. Making a function synchronous for this reason violates the class independence. Can we still justify keeping this function synchronous?